common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -652,6 +829,274 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         )
     }
 
+    @VisibleForTesting
+    fun performSignUpStartRequest(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpStartCommandParameters
+    ): SignUpStartApiResult {
+        return oAuth2Strategy.performSignUpStart(
+            commandParameters = parameters
+        )
+    }
+
+    @VisibleForTesting
+    fun performSignUpStartUsingPasswordRequest(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpStartUsingPasswordCommandParameters
+    ): SignUpStartApiResult {
+        return oAuth2Strategy.performSignUpStartUsingPassword(
+            commandParameters = parameters
+        )
+    }
+
+    private fun performSignUpChallengeCall(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        signupToken: String
+    ): SignUpChallengeApiResult {
+        return oAuth2Strategy.performSignUpChallenge(signUpToken = signupToken)
+    }
+
+    private fun performSignUpSubmitCode(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpSubmitCodeCommandParameters
+    ): SignUpContinueApiResult {
+        return oAuth2Strategy.performSignUpSubmitCode(commandParameters = parameters)
+    }
+    private fun performSignUpSubmitPassword(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpSubmitPasswordCommandParameters
+    ): SignUpContinueApiResult {
+        return oAuth2Strategy.performSignUpSubmitPassword(commandParameters = parameters)
+    }
+
+    @VisibleForTesting
+    fun performSignUpSubmitUserAttributes(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: SignUpSubmitUserAttributesCommandParameters
+    ): SignUpContinueApiResult {
+        return oAuth2Strategy.performSignUpSubmitUserAttributes(commandParameters = parameters)
+    }
+
+    private fun SignUpChallengeApiResult.toSignUpStartCommandResult(): SignUpStartCommandResult {
```

Comment:
Please add some comments describing why this method is needed

Reply:
Done